kornia/geometry/liegroup/se3.py

```diff
@@ -0,0 +1,215 @@
+            - 0.5 * omega_hat
+            + ((1 - theta * (theta / 2).cos() / (2 * (theta / 2).sin())) / theta.pow(2))[..., None]
+            * (omega_hat @ omega_hat)
+        )
+        t = where(theta != 0.0, (self.t.reshape(-1, 1, 3) * V_inv).sum(-1), self.t)
+        return concatenate((t, omega), -1)
+
+    @staticmethod
+    def hat(v) -> Tensor:
+        """Converts elements from vector space to lie algebra. Returns matrix of shape :math:`(B,4,4)`.
+
+        Args:
+            v: vector of shape :math:`(B,6)`.
+
+        Example:
+            >>> v = torch.ones((1,6))
+            >>> m = Se3.hat(v)
+            >>> m
+            tensor([[[ 0.,  0., -1.,  1.],
+                     [ 0.,  1.,  0., -1.],
+                     [ 0., -1.,  1.,  0.],
+                     [ 0.,  1.,  1.,  1.]]])
+        """
+        KORNIA_CHECK_SHAPE(v, ["B", "6"])
+        t = v[..., 0:3].reshape(-1, 1, 3)
+        omega = v[..., 3:]
+        rt = concatenate((So3.hat(omega), t.reshape(-1, 1, 3)), 1)
+        # TODO: infer device, dtype
+        return concatenate((zeros(v.shape[0], 4, 1).to(rt.device, rt.dtype), rt), -1)
+
+    @staticmethod
+    def vee(omega) -> Tensor:
+        """Converts elements from lie algebra to vector space.
+
+        Args:
+            omega: 4x4-matrix representing lie algebra of shape :math:`(B,4,4)`.
+
+        Returns:
+            vector of shape :math:`(B,6)`.
+
+        Example:
+            >>> v = torch.ones((1,6))
+            >>> omega_hat = Se3.hat(v)
+            >>> Se3.vee(omega_hat)
+            tensor([[1., 1., 1., 1., 1., 1.]])
+        """
+        KORNIA_CHECK_SHAPE(omega, ["B", "4", "4"])
+        t = omega[..., 3, 1:]
+        v = So3.vee(omega[..., 0:3, 1:])
+        return concatenate((t, v), 1)
+
+    @classmethod
+    def identity(cls, batch_size: int, device=None, dtype=None) -> 'Se3':
+        """Create a Se3 group representing an identity rotation and zero translation.
+
+        Args:
+            batch_size: the batch size of the underlying data.
+
+        Example:
+            >>> s = Se3.identity(batch_size=2)
+            >>> s.data
+            rotation: real: tensor([[1.],
+                    [1.]], grad_fn=<SliceBackward0>)
+            vec: tensor([[0., 0., 0.],
+                    [0., 0., 0.]], grad_fn=<SliceBackward0>)
+            translation: tensor([[0., 0., 0.],
+                    [0., 0., 0.]])
+        """
+        t: Tensor = as_tensor([0.0, 0.0, 0.0])
+        t = t.repeat(batch_size, 1)
+        return cls(So3.identity(batch_size, device, dtype), t.to(device, dtype))
+
+    def matrix(self) -> Tensor:
+        """Returns the matrix representation of shape :math:`(B, 4, 4)`.
+
+        Example:
+            >>> s = Se3(So3.identity(batch_size=1), torch.ones((1,3)))
+            >>> s
+            tensor([[[0., 1., 0., 0.],
+                     [0., 0., 1., 0.],
+                     [0., 0., 0., 1.],
+                     [0., 1., 1., 1.]]], grad_fn=<CatBackward0>)
```

Comment:
This output is not correct. Is there a test for this ?

Reply:
I've corrected this issue. Yes a test is there for matrix. Please check